logbook-jaxrs/src/main/java/org/zalando/logbook/jaxrs/LocalResponse.java

```diff
@@ -9,18 +10,89 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
+@AllArgsConstructor
 final class LocalResponse implements HttpResponse {
 
-    private final ContainerResponseContext context;
+    private interface State {
+
+        default State with() {
+            return this;
+        }
+
+        default State without() {
+            return this;
+        }
+
+        default State buffer(final ContainerResponseContext context) {
+            return this;
+        }
+
+        default byte[] getBody() {
+            return new byte[0];
+        }
+
+    }
+
+    private static final class Unbuffered implements State {
+
+        @Override
+        public State with() {
+            return new Offering();
+        }
+
+    }
+
+    private static final class Offering implements State {
+
+        @Override
+        public State without() {
+            return new Unbuffered();
+        }
+
+        @Override
+        public State buffer(final ContainerResponseContext context) {
+            final TeeOutputStream stream = new TeeOutputStream(context.getEntityStream());
+            context.setEntityStream(stream);
+            return new Buffering(stream);
+        }
+
+    }
 
-    private TeeOutputStream stream;
-    private byte[] body;
+    @AllArgsConstructor
+    private static final class Buffering implements State {
+
+        private final TeeOutputStream stream;
+
+        @Override
+        public State without() {
+            return new Ignoring(stream);
+        }
+
+        @Override
+        public byte[] getBody() {
+            return stream.toByteArray();
+        }
+
+    }
+
+    @AllArgsConstructor
+    private static final class Ignoring implements State {
+
+        private final TeeOutputStream stream;
+
+        @Override
+        public State with() {
+            return new Buffering(stream);
+        }
 
-    public LocalResponse(final ContainerResponseContext context) {
-        this.context = context;
     }
 
+    private final AtomicReference<State> state = new AtomicReference<>(new Unbuffered());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/whiskeysierra/logbook/pullRequest?prid=4506178)